articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+- Status might have any of the following values:
+  * password_matched: the password you provided for that user matched
+  * invalid_password: the password you provided haven't matched
+  * user_not_found: the user wasn't available in the connection database you specified.
+
+Keep in mind that we try to recover from most of the errors instead of failing the job, so even when the job is complete
+there might be rows that have had errors and so they are not included in the results. You should check the errors (see below) to get more information about what have happened.
+
+```
+*IMPORTANT:* To keep a high security standard you must delete the job once you have the results and store them securely. The results will be automatically deleted after some time of completing / failing the job. Once the job has been processed the input files are deleted automatically and you will be not able to access them.
+```
+
+### Querying errors
+If we found errors processing the job they will be made available to you through this endpoint. Keep in mind that there might be errors even in a non-failed (completed) job; that's because if we find an error that can be scoped to an specific record, we keep processing the rest of the input file instead of completely failing the job.
+
+To query the errors you should make `GET` request to `/api/{job_id}/errors`. You will get a JSON array result similar to:
+
+```json
+[
+  {
+    "row": {
+      "username": "john.doe"
+    },
+    "errorMessage": "The plain_text_password property is required",
+    "errorCode": "row_validation_error"
+  },
+  {
+    "row": {
+      "email": "john.doe@example.com"
+    },
+    "errorMessage": "The plain_text_password property is required",
+    "errorCode": "row_validation_error"
+  },
+  {
+    "row": {
+      "plain_text_password": "\<REMOVED\>"
+    },
+    "errorMessage": "Username or email are required",
+    "errorCode": "row_validation_error"
+  }
+]
+```
+
+- `row` property contains information about the failing row, if it included a password it will be replaced by the tag "\<REMOVED\>"; it might also contain username or email depending on what you provided.
+
+- `errorMessage` an human readable description of the error.
+
+- `errorCode`: an invariant code of the error; it might be:
+  * row_validation_error: when the row does not match a valid format.
+  * unknown_error: when the error is scoped to the row but we cannot classify it over a more specific type.
+
+### Deleting input and results data
+Once you are done, you have the results, the errors, etc. it is advisable to delete the input data, the job, the results and the errors. To do so you should make a `DELETE` request to `/api/jobs/{job_id}`. Make sure you have a `delete:passwords_checking_job` scope. If everything went well you will get back an empty `204` response. You can check that everything has been deleted trying to get the results for the job using the endpoint described above.
```

Comment:
Once you have retrieved the results, make a `DELETE` request to `/api/jobs/{job_id}`. Make sure you have a `delete:passwords_checking_job` scope. You should get back an empty `204` response. You can check that everything has been deleted trying to get the results for the job using the endpoint described above.
 +